pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -50,12 +50,38 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 		}
 	}
 
+	allErrs = append(allErrs, validateZones(p.Zones, ci.Zones, fldPath.Child("zones"))...)
 	allErrs = append(allErrs, validateUUIDV4s(p.AdditionalNetworkIDs, fldPath.Child("additionalNetworkIDs"))...)
 	allErrs = append(allErrs, validateUUIDV4s(p.AdditionalSecurityGroupIDs, fldPath.Child("additionalSecurityGroupIDs"))...)
 
 	return allErrs
 }
 
+func validateZones(input []string, actual []string, fldPath *field.Path) field.ErrorList {
```

Comment:
```suggestion
func validateZones(input []string, available []string, fldPath *field.Path) field.ErrorList {
```

actual is not meaningful as an argument to validateZones